src/components/Posts/ExpandedPostItem.tsx

```diff
@@ -0,0 +1,230 @@
+    alignItems: "center",
+  },
+  backButton: {
+    backgroundColor: "transparent",
+    paddingRight: 10,
+  },
+  headerTitle: {
+    fontWeight: "bold",
+  },
+};
+
+const ExpandedPostItem = () => {
+  const dispatch = useAppDispatch();
+  const user = useAppSelector((state) => state.user);
+  const post = useAppSelector((state) => state.post);
+  const urlParams = useParams();
+
+  useEffect(() => {
+    const updatedExpandedPost = async () => {
+      const backupFetch = await axios.get(
+        "http://localhost:3001/api/posts/fetchPost",
+        {
+          params: {
+            userId: user.userId,
+            postId: urlParams.postId,
+          },
+        }
+      );
+      dispatch(setExpandedPost(backupFetch.data));
+    };
+    updatedExpandedPost();
+  }, [dispatch, user.userId, urlParams.postId]);
+
+  const likePost = async (postId: number, userId?: number) => {
+    await axios.post("http://localhost:3001/api/posts/likePost", {
+      postId,
+      userId,
+    });
+    const updatedPost = {
+      ...post,
+      isLikedByCurrentUser: !post.isLikedByCurrentUser,
+      numberOfLikes: post.isLikedByCurrentUser
+        ? post.numberOfLikes - 1
+        : post.numberOfLikes + 1,
+    };
+    dispatch(updateExpandedPost(updatedPost));
+  };
+
+  const unlikePost = async (postId: number, userId?: number) => {
+    await axios.delete("http://localhost:3001/api/posts/unlikePost", {
+      params: {
+        postId,
+        userId,
+      },
+    });
+    const updatedPost = {
+      ...post,
+      isLikedByCurrentUser: !post.isLikedByCurrentUser,
+      numberOfLikes: post.isLikedByCurrentUser
+        ? post.numberOfLikes - 1
+        : post.numberOfLikes + 1,
+    };
+    dispatch(updateExpandedPost(updatedPost));
+  };
+
+  const navigate = useNavigate();
+
+  return (
+    <Card sx={styles.card}>
+      <Box style={styles.topHeader}>
+        <IconButton style={styles.backButton} onClick={() => navigate(-1)}>
+          <KeyboardBackspaceIcon color="secondary" />
+        </IconButton>
+        <Typography style={styles.headerTitle}>Post</Typography>
+      </Box>
+      <CardHeader
+        avatar={<Avatar>CK</Avatar>}
+        action={
+          <IconButton>
+            <MoreVertIcon />
+          </IconButton>
+        }
+        title={`${post.displayName} `}
+        subheader={`@${post.username}`}
+      />
+      <CardContent sx={{ width: 400 }}>
+        <Typography>{post.textContent}</Typography>
+      </CardContent>
+      {post.imagePath && (
+        <CardMedia
+          sx={{ maxWidth: 200, margin: "auto" }}
+          component="img"
+          image={post.imagePath}
+        />
+      )}
+      <Box sx={styles.timestampBox}>
+        <Typography component={"span"} sx={styles.timestamp}>
+          {post.timestamp}
+        </Typography>
+      </Box>
+      <Divider variant="middle" />
+      <Box sx={styles.actionData}>
+        <Button component={"span"} sx={styles.actionNumbers}>
```

Comment:
I have a few points:

1. Was there a reason to make the button a span? Can we remove that and make it a normal button? Or a div?
2. Since the number and action are still in different containers, we're still getting that bug where the numbers shift when they change. I think the fix is to do something like:
```
<Button>
  <Typography>{`${post.numberOfLikes} Likes`}</Typography>
</Button>
```

Reply:
1. Tbh I can't remember why I did that, but it works without it so I removed it
2. The update made was so that it doesn't shift the entire right, but if we're talking about the shift within the "Likes" container for example I don't think that solves it. I tried to fix it with this commit:

https://github.com/TweetClone/twitter-clone-frontend/pull/41/commits/3fa8a6fd82e163e2900d8720bcdd246c25024fff

Reply:
As discussed, we'll revert this to the way you originally had it and make a new ticket to address this bug